AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -19,13 +25,58 @@ class DoubleVideoViewController: UIViewController, CellTitled {
 
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
+        imagePickerController.delegate = self
+        imagePickerController.sourceType = .photoLibrary
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
     }
 
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        present(imagePickerController, animated: true, completion: nil)
+    }
+    
+    //MARK:- ImagePicker Delegate Method
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        
+        if let url = info[UIImagePickerControllerMediaURL] as? URL{
+            self.dismiss(animated: true){
+                self.playSelectedVideo(url: url)
+            }
+        }
+    }
     
+    //MARK:- Utilities
+    func playSelectedVideo(url: URL){
+        let playerItem: AVPlayerItem = AVPlayerItem(url: url)
+        let playerLayer: AVPlayerLayer = AVPlayerLayer()
+        
+        if videoContainerTop.layer.sublayers == nil || topPlayer.rate == 0.0 {
```

Comment:
It's kind of a perfect submission so there's little to say. I don't see evidence of the third bonus, the cryptic "use a datastructure" to manage players and views. I was looking for arrays. In such a case this logic could be refactored to be less verbose but also might need to be changed a bit to work generally. Then you could have n players. I don't know why...